Removes minimum font size from reaction rendering

bp-kelley opened the pull request:
Fixes #8195 

Before:

![image](https://github.com/user-attachments/assets/a6543628-787b-4a4b-8f23-bd645bf1dd54)

After:

![image](https://github.com/user-attachments/assets/87262012-dd39-428e-a804-93faf68d723a)

It's still not perfect as the text can now get cutoff, but it does look a tad better.

bp-kelley commented:
Here is an extreme case:

![image](https://github.com/user-attachments/assets/3a8b59cc-d999-4446-9fbb-0c56db3fb062)

bp-kelley commented:
Test Mol block

```
data = """$RXN

      Mrv2019  070820240955

  2  1
$MOL

  Mrv2019 07082409552D          

  9  9  0  0  0  0            999 V2000
   -7.2098    0.4625    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
   -7.9243    0.0500    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
   -7.9243   -0.7750    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
   -7.2098   -1.1875    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
   -6.4954   -0.7750    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
   -6.4954    0.0500    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
   -5.7809    0.4625    0.0000 C   0  0  0  0  0  0  0  0  0  1  0  0
   -5.7809    1.2875    0.0000 O   0  0  0  0  0  0  0  0  0  0  0  0
   -5.0664    0.0500    0.0000 O   0  0  0  0  0  0  0  0  0  0  0  0
  1  2  1  0  0  0  0
  2  3  2  0  0  0  0
  3  4  1  0  0  0  0
  4  5  2  0  0  0  0
  5  6  1  0  0  0  0
  1  6  2  0  0  0  0
  6  7  1  0  0  0  0
  7  8  2  0  0  0  0
  7  9  1  0  0  0  0
M  END
$MOL

  Mrv2019 07082409552D          

  4  3  2  0  0  0            999 V2000
   -2.1041    0.7645    0.0000 H   0  0  0  0  0  0  0  0  0  7  0  0
   -2.5166    0.0500    0.0000 N   0  0  0  0  0  0  0  0  0  4  0  0
   -2.1041   -0.6644    0.0000 L   0  0  0  0  0  0  0  0  0  5  0  0
   -3.3416    0.0500    0.0000 L   0  0  0  0  0  0  0  0  0  6  0  0
  1  2  1  0  0  0  0
  2  3  1  0  0  0  0
  2  4  1  0  0  0  0
  3 F    2   1   6
  4 F    2   1   6
M  ALS   3  2 F H   C   
M  ALS   4  2 F H   C   
M  END
$MOL

  Mrv2019 07082409552D          

 11 11  2  0  0  0            999 V2000
    3.3036    0.0500    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
    4.0180    0.4625    0.0000 C   0  0  0  0  0  0  0  0  0  1  0  0
    4.7325    0.0500    0.0000 N   0  0  0  0  0  0  0  0  0  4  0  0
    4.0180    1.2875    0.0000 O   0  0  0  0  0  0  0  0  0  0  0  0
    2.5891    0.4625    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
    1.8746    0.0500    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
    1.8746   -0.7750    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
    2.5891   -1.1875    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
    3.3035   -0.7750    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
    5.4470    0.4625    0.0000 L   0  0  0  0  0  0  0  0  0  5  0  0
    4.7325   -0.7750    0.0000 L   0  0  0  0  0  0  0  0  0  6  0  0
  1  2  1  0  0  0  0
  2  3  1  0  0  0  0
  2  4  2  0  0  0  0
  5  6  2  0  0  0  0
  6  7  1  0  0  0  0
  7  8  2  0  0  0  0
  8  9  1  0  0  0  0
  1  5  1  0  0  0  0
  1  9  2  0  0  0  0
  3 10  1  0  0  0  0
  3 11  1  0  0  0  0
 10 F    2   1   6
 11 F    2   1   6
M  ALS  10  2 F H   C   
M  ALS  11  2 F H   C   
M  END
"""
```

greglandrum reviewed (changes requested):
Please add a test for this

bp-kelley commented:
@greglandrum @DavidACosgrove it turns out the reaction draw tests didn't actually fail when the depictions were different.  I've updated the results with the new renderings (after visual inspection) and added png hashes when FREETYPE support is not turned on, they were missing before.

So the tests: 
   CAIRO = on/off  not that exciting, turns off png
   FREETYPE on/off - affects png and sag apparently

DavidACosgrove commented:
@bp-kelley These are old tests, we rely on a visual inspection.  There's a script $RDBASE/Code/GraphMol/MolDraw2D/side_by_side_images.py that reads the same image files from 2 different directories and puts them in an HTML table.  I generally use that at the end.
The hash codes don't work for PNGs - it only takes a single tiny difference between 2 floats and you get a different hash code and that's pretty much always the case on different systems.  They are more reliable with SVGs because, being ASCII, we control the number of decimal places in the coords and the inter-system differences are much fewer.  That's why they aren't put in for the PNG tests.  Mostly we inspect the innards of the SVGs to check whether a change has been successful and assume the PNG will look the same since they use the same underlying drawing code.

bp-kelley commented:
In either case it was computing the hash but not actually testing it, so anything would be better!----Brian KelleyOn Jan 25, 2025, at 02:50, David Cosgrove ***@***.***> wrote:﻿
@bp-kelley These are old tests, we rely on a visual inspection.  There's a script $RDBASE/Code/GraphMol/MolDraw2D/side_by_side_images.py that reads the same image files from 2 different directories and puts them in an HTML table.  I generally use that at the end.
The hash codes don't work for PNGs - it only takes a single tiny difference between 2 floats and you get a different hash code and that's pretty much always the case on different systems.  They are more reliable with SVGs because, being ASCII, we control the number of decimal places in the coords and the inter-system differences are much fewer.  That's why they aren't put in for the PNG tests.  Mostly we inspect the innards of the SVGs to check whether a change has been successful and assume the PNG will look the same since they use the same underlying drawing code.

—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you were mentioned.Message ID: ***@***.***>

greglandrum commented:
Superseded by #8211
